torch/utils/data/datapipes/iter/combinatorics.py

```diff
@@ -86,24 +85,24 @@ def __init__(self,
         else:
             self.datapipe = datapipe.unbatch(unbatch_level=unbatch_level)
         self.buffer_size = buffer_size
-        self._buffer = []
 
-    def buffer_replace(self, x):
-        idx = random.randint(0, self.buffer_size - 1)
-        val = self._buffer[idx]
-        self._buffer[idx] = x
+    @staticmethod
```

Comment:
Convert it to staticmethod because nothing from `self` instance is needed for this function.